lib/inference_engines/runner/step_execution.rb

```diff
@@ -130,17 +132,27 @@ def select_asset(uuids)
         FactChanges.new.tap do |updates|
           assets = uuids.map { |uuid| Asset.find_by(uuid: uuid) }
           updates.add_assets(@step.asset_group.id, uuids)
-          #@step.asset_group.add_assets(assets)
+          # @step.asset_group.add_assets(assets)
         end
       end
 
       def unselect_asset(uuids)
         FactChanges.new.tap do |updates|
           assets = uuids.map { |uuid| Asset.find_by(uuid: uuid) }
           updates.remove_assets(@step.asset_group.id, uuids)
-          #@step.asset_group.remove_assets(assets)
+          # @step.asset_group.remove_assets(assets)
         end
       end
+
+      private
+
+      def input_url
```

Comment:
In the cases I looked at we actuall did a regex on these urls and extracted the id from that. So in the event we decide to keep other activities in separate processes, there is grounds for simplification here.